Tests/scripts/hook_validations/script.py

```diff
@@ -53,13 +53,50 @@ def _is_sub_set(cls, supposed_bigger_list, supposed_smaller_list):
 
     def is_backward_compatible(self):
         """Check if the script is backward compatible."""
-        self.is_arg_changed()
+        if not self.old_script:
+            return True
+
         self.is_context_path_changed()
         self.is_docker_image_changed()
+        self.is_added_required_args()
+        self.is_arg_changed()
         self.is_there_duplicates_args()
 
         return self._is_valid
 
+    @classmethod
+    def _get_arg_to_required_dict(cls, script_json):
+        """Get a dictionary arg name to its required status.
+
+        Args:
+            script_json (dict): Dictionary of the examined script.
+
+        Returns:
+            dict. arg name to its required status.
+        """
+        arg_to_required = {}
+        args = script_json.get('args', [])
+        for arg in args:
+            arg_to_required[arg.get('name')] = arg.get('required', False)
+
+        return arg_to_required
+
+    def is_added_required_args(self):
+        """Check if required arg were added."""
+        current_args_to_required = self._get_arg_to_required_dict(self.current_script)
+        old_args_to_required = self._get_arg_to_required_dict(self.old_script)
+
+        for arg, required in current_args_to_required.items():
+            if required:
+                if (arg not in old_args_to_required) or \
+                        (arg in old_args_to_required and required != old_args_to_required[arg]):
+                    print_error("You've added required args in the script file '{}', the field is '{}'".format(
+                        self.file_path, arg))
+                    self._is_valid = False
```

Comment:
please move the `self._is_valid = False` to the caller function that dose that anyway